sdk/core/Azure.Core.TestFramework/src/Instrumentation/InstrumentResultInterceptor.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Castle.DynamicProxy;
+
+namespace Azure.Core.TestFramework
+{
+    internal class InstrumentResultInterceptor : IInterceptor
+    {
+        private static readonly MethodInfo InstrumentOperationInterceptorMethodInfo = typeof(InstrumentResultInterceptor).
+            GetMethod(nameof(InstrumentOperationInterceptor), BindingFlags.NonPublic | BindingFlags.Instance);
+
+        private readonly ClientTestBase _testBase;
+
+        public InstrumentResultInterceptor(ClientTestBase testBase)
+        {
+            _testBase = testBase;
+        }
+
+        public void Intercept(IInvocation invocation)
+        {
+            var type = invocation.Method.ReturnType;
+
+            // We don't want to instrument generated rest clients.
```

Comment:
What about for LLCs, or is the assumption that we wouldn't be writing recorded tests with them?

Reply:
LLCs won't have `GetNestedClient()` methods that this interceptor is handling.